### PR TITLE
Added unique and uniqueBy transform stream functions

### PR DESCRIPTION
**Pull request by Menshawi**

### CODE REVIEW

File: lib/unique.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * @returns {void}
+ */
+
+function createUniqueStream(){
+
+  const cache = [];
+
+  function transform( item, _, next ){
+    
+    const emit = cache.indexOf( item ) > -1
```
Comment (noisygerman) [inline lib/unique.js]: cache.includes( item )

File: lib/unique.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      ? this.emit.bind( this, DuplicateEventKey )
+      : this.push.bind( this );
+
+    cache.push( item );
```
Comment (noisygerman) [inline lib/unique.js]: why'd you push it if it already exists?

File: lib/uniqueBy.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+function createUniqueByStream( iteratee ){
+
+  assert( isFunction( iteratee ) || isString( iteratee ), 'Iteratee can be eitehr a function or a string' );
+
+  const cache = new Set();
+
+  function transform( item, _, next ){
+    
+
+    const v = isFunction( iteratee )
```
Comment (noisygerman) [inline lib/uniqueBy.js]: look at `https://lodash.com/docs/4.17.15#iteratee` and used it instead of the assert above

**Review by noisygerman — CHANGES_REQUESTED** [review]

File: lib/unique.js (hunk 2 above)
Comment (Menshawi) [reply lib/unique.js]: I'll just use a set then

**Review by noisygerman — APPROVED** [review]

**Pull request merged**